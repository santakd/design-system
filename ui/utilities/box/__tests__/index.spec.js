/* eslint-env jest */
import React from 'react';
import createHelpers from '../../../../jest.helpers';
import { mount } from 'enzyme';
const { matchesMarkup } = createHelpers(__dirname);

describe('renders box utilities', () => {
  it('renders box', () =>
    matchesMarkup(
      <div className="slds-box">
        <p>This is a regular-sized box.</p>
      </div>
    ));

  it('renders xx small box', () =>
    matchesMarkup(
      <div className="slds-box slds-box_xx-small">
        <p>This is a regular-sized box.</p>
      </div>
    ));

  it('renders x small box', () =>
    matchesMarkup(
      <div className="slds-box slds-box_x-small">
        <p>This is a regular-sized box.</p>
      </div>
    ));

  it('renders small box', () =>
    matchesMarkup(
      <div className="slds-box slds-box_small">
        <p>This is a regular-sized box.</p>
      </div>
    ));

  it('renders actionable box with interactions', () => {
    const el = mount(
      <a
        href="javascript:void(0);"
        className="slds-box slds-box_link"
        style={{ display: 'block' }}
      >
        <p>This is an actionable box with interactions.</p>
      </a>
    );
    return matchesMarkup(
      el
        .find('.slds-box_link')
        .at(0)
        .simulate('mouseEnter')
        .html()
    );
  });
});
