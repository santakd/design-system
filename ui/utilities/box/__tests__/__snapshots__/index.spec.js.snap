// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders box utilities renders actionable box with interactions 1`] = `"<a href=\\"javascript:void(0);\\" class=\\"slds-box slds-box_link\\" style=\\"display: block;\\"><p>This is an actionable box with interactions.</p></a>"`;

exports[`renders box utilities renders box 1`] = `
<div
  className="slds-box"
>
  <p>
    This is a regular-sized box.
  </p>
</div>
`;

exports[`renders box utilities renders small box 1`] = `
<div
  className="slds-box slds-box_small"
>
  <p>
    This is a regular-sized box.
  </p>
</div>
`;

exports[`renders box utilities renders x small box 1`] = `
<div
  className="slds-box slds-box_x-small"
>
  <p>
    This is a regular-sized box.
  </p>
</div>
`;

exports[`renders box utilities renders xx small box 1`] = `
<div
  className="slds-box slds-box_xx-small"
>
  <p>
    This is a regular-sized box.
  </p>
</div>
`;
